feat: added interactive tag (#15808)
carbon-design-system/carbon: packages/react/src/components/Tag/InteractiveTag.mdx

import { Canvas, Story } from '@storybook/blocks';
import * as InteractiveTagStories from './InteractiveTag.stories';

# Interactive Tag

[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Tag)
&nbsp;|&nbsp;
[Usage guidelines](https://www.carbondesignsystem.com/components/tag/usage)
&nbsp;|&nbsp;
[Accessibility](https://www.carbondesignsystem.com/components/tag/accessibility)

## Overview

> Multiple or single tags can be used to categorize items.
> 
> Use short labels for easy scanning. Use two words only if necessary to describe
> the status and differentiate it from other tags.

Reviewer: ```suggestion
Tags can be used to categorize items. Use short labels for easy scanning. Use two words only if necessary to describe the status and differentiate it from other tags.
```

## Dismissible

Dismissible tags are used to remove tags that can be filtered out by the user.

When to use:

- Use DismissibleTag when you want to allow users to easily clear or dismiss
  specific filters applied to items.

<Canvas>
  <Story of={InteractiveTagStories.Dismissible} />
</Canvas>

```jsx
<DismissibleTag type="red" className="some-class" title="Clear Filter">
  {'Tag content'}
</DismissibleTag>
```

## Selectable

> Selectable tags are used to select or unselect single or multiple items in a
> form situation. Selectable tags can also be used in filtering scenarios by label
> functionality when they do not need a dimsdale functionality.

Reviewer: ```suggestion
Selectable tags are used to select or unselect single or multiple items. Selectable tags can also be used in filtering by label scenarios when they do not need to be dismissed.
```

When to use:

- Use selectable tags to toggle between a selected or unselected state.

> - Use when needing to filter without an explicit dismissable functionality. (We

Reviewer: ```suggestion
- Use when needing to filter without explicit dismissible functionality. (We
```

  need to be a little more specific around filtering for selectable versus

>   dismissable tags).

Reviewer: ```suggestion
  dismissible tags).
```

<Canvas>
  <Story of={InteractiveTagStories.Selectable} />
</Canvas>

```jsx
<SelectableTag className="some-class" title="Clear Filter">
  {'Tag content'}
</SelectableTag>
```

## Operational

Operational tags enable the user to view a list of all items associated with a
given tag in different ways.

When to use:

> - Use to view a list of items with the same tag in a tagclip, popover, or

Reviewer: Is this supposed to be `Toggletip`?

  breadcrumb detail view.

When not to use:

- Do not use operational tags as a replacement for links that direct you to an
  entirely different page or launch you out to another tab.

> - Do not use in combination with Dismissable tags. Instead, consider letting the

Reviewer: ```suggestion
- Do not use it in combination with `DismissibleTag`. Instead, consider letting the
```
